sql/core/src/main/java/org/apache/spark/sql/execution/RecordBinaryComparator.java

```diff
@@ -49,9 +51,13 @@ public int compare(
     // for architectures that support unaligned accesses, chew it up 8 bytes at a time
     if (Platform.unaligned() || (((leftOff + i) % 8 == 0) && ((rightOff + i) % 8 == 0))) {
       while (i <= leftLen - 8) {
-        final long v1 = Platform.getLong(leftObj, leftOff + i);
-        final long v2 = Platform.getLong(rightObj, rightOff + i);
+        long v1 = Platform.getLong(leftObj, leftOff + i);
+        long v2 = Platform.getLong(rightObj, rightOff + i);
         if (v1 != v2) {
+          if (ByteOrder.nativeOrder().equals(ByteOrder.LITTLE_ENDIAN)) {
+            v1 = Long.reverseBytes(v1);
+            v2 = Long.reverseBytes(v2);
+          }
           return v1 > v2 ? 1 : -1;
```

Comment:
Pulling together my comment with https://github.com/apache/spark/pull/25491#issuecomment-551957225 - maybe this also has to be `Long.compareUnsigned`?

Reply:
Yes, thanks for point out this. This is the same issue with https://github.com/apache/spark/pull/25491#issuecomment-551957225.  And this can be replicated in our cluster.